bundles/org.openhab.binding.philipsair/src/main/java/org/openhab/binding/philipsair/internal/PhilipsAirHandler.java

```diff
@@ -0,0 +1,409 @@
+import static org.openhab.binding.philipsair.internal.PhilipsAirBindingConstants.WATER_LEVEL;
+import static org.openhab.binding.philipsair.internal.PhilipsAirBindingConstants.WICKS_FILTER;
+
+import java.io.UnsupportedEncodingException;
+import java.math.BigDecimal;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.List;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.measure.quantity.Dimensionless;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.smarthome.core.library.dimension.Density;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerCallback;
+import org.eclipse.smarthome.core.thing.type.ChannelKind;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.philipsair.internal.connection.PhilipsAirAPIConnection;
+import org.openhab.binding.philipsair.internal.connection.PhilipsAirAPIException;
+import org.openhab.binding.philipsair.internal.model.PhilipsAirPurifierDataDTO;
+import org.openhab.binding.philipsair.internal.model.PhilipsAirPurifierDeviceDTO;
+import org.openhab.binding.philipsair.internal.model.PhilipsAirPurifierFiltersDTO;
+import org.openhab.binding.philipsair.internal.model.PhilipsAirPurifierWritableDataDTO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link PhilipsAirHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Michal Boronski - Initial contribution
+ */
+@NonNullByDefault
+public class PhilipsAirHandler extends BaseThingHandler {
+    private static final long INITIAL_DELAY_IN_SECONDS = 5;
+    private final Logger logger = LoggerFactory.getLogger(PhilipsAirHandler.class);
+    private @Nullable ScheduledFuture<?> refreshJob;
+    private @NonNullByDefault({}) PhilipsAirAPIConnection connection;

```

Comment:
There is a very real possibility that this could be null during normal execution.
```suggestion
    private @Nullable PhilipsAirAPIConnection connection;

```